docs/specification/draft/basic/authentication.md

```diff
@@ -0,0 +1,385 @@
+transport **SHOULD** conform to this specification, while those using alternative
+transports **MUST** follow established security best practices for their protocol.
+
+For deployments supporting third-party clients via HTTP+SSE, implementations **SHOULD**
+expect authentication as defined here. Controlled environment deployments like corporate
+networks **MAY** implement alternative methods such as certificate-based authentication.
+
+### 1.3 Standards Compliance
+
+This authentication mechanism is based on established specifications listed below, but
+implements a selected subset of their features to ensure security and interoperability
+while maintaining simplicity:
+
+- [OAuth 2.1 IETF DRAFT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12)
+- OAuth 2.0 Authentication Server Metadata
+  ([RFC8414](https://datatracker.ietf.org/doc/html/rfc8414))
+- OAuth 2.0 Dynamic Client Registration Protocol
+  ([RFC7591](https://datatracker.ietf.org/doc/html/rfc7591))
+
+## 2. Authentication Flow
+
+### 2.1 Overview
+
+MCP is designed to support interactive, user-facing developer tools that connect to
+arbitrary servers. This leads to two key design decisions:
+
+1. All MCP implementations **MUST** implement OAuth 2.1 with appropriate security
+   measures for both confidential and public clients.
+
+2. All MCP implementations **SHOULD** support the OAuth 2.0 Dynamic Client Registration
+   Protocol ([RFC7591](https://datatracker.ietf.org/doc/html/rfc7591)).
+
+3. All MCP implementations **SHOULD** implement OAuth 2.0 Authentication Server Metadata
+   ([RFC8414](https://datatracker.ietf.org/doc/html/rfc8414)). Servers that do not
+   support Authentication Server Metadata **MUST** follow the default URI schema.
+
+### 2.2 Basic OAuth 2.1 Authentication
+
+When authentication is required, servers **MUST** respond with HTTP 401 Unauthorized. The
+basic
+[OAuth 2.1 IETF DRAFT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12)
+authentication flow with PKCE comprises the following sequence:
+
+```mermaid
+sequenceDiagram
+    participant B as User-Agent (Browser)
+    participant C as Client
+    participant M as MCP Server
+
+    C->>M: MCP Request
+    M->>C: HTTP 401 Unauthorized
+    Note over C: Generate code_verifier and code_challenge
+    C->>B: Open browser with authorization URL + code_challenge
+    B->>M: GET /authorize
+    Note over M: User logs in and authorizes
+    M->>B: Redirect to callback URL with auth code
+    B->>C: Callback with authorization code
+    C->>M: Token Request with code + code_verifier
+    M->>C: Access Token + Refresh Token
+    C->>M: MCP Request with Access Token
+    Note over C,M: Begin standard MCP message exchange
+```
+
+### 2.3 Server Metadata Discovery
+
+For server capability discovery, MCP clients _SHOULD_ follow the OAuth 2.0 Authorization
+Server Metadata protocol defined in
+[RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
```

Comment:
I suggest adding a separate paragraph which emphasizes that this is optional for servers.